docs: introduce global theming sooner (#937)
chakra-ui/chakra-ui: website/docs/theming/overview.mdx

and changing project is a challenging task. To solve this, we need to have a
simple and consistent system.

> Chakra provides a unified component-driven CSS API that it builds it's own components

Reviewer: ```suggestion
Chakra provides a unified component-driven CSS API that it builds its own components
```

  schemes. For example, an outline button with a red color scheme.
- **Color mode.** A component can change its visual styles based on color mode
  (light or dark).

>   

Author: Not sure if it's an appropriate way to look at theming but it feels like you can theme "locally" or "globally".

Reviewer: I like this distinction. @segunadebayo thoughts?

Reviewer 2: Yes, it make sense

## Theming local and global

- **Local.** Custom components can be themed locally using a `styled`-like API
- **Global.** Both Chakra and Custom components can be themed via the central chakra theme using `themeKey`

### Local Theming

In styled components, you usually do this:

```js live=false
const MyComponent = styled("div")({
  // any style you can imagine goes here
})

> ```

Author: I change this example to not use `Button` because it's a bit confusing, since likely they'll just customize the existing out-of-the-box button instead of making a new one. I think it's more likely that they'll make a new component that is not out-of-the-box from Chakra, so I just use `MyComponent` to represent that.

styles consistent and extensible.

```js live=false
const MyComponent = chakra("div", {
  // base or default styles
  baseStyle: {...},
  // styles for the different sizes
  sizes: {...},
  // styles for the different visual styles
  variants: {...},
  // attach props
  attrs: {...}
})
```

### Global Theming

The above is convenient for quickly creating custom themed components, but...

**What if you want to maintain your theme centrally?**

**What if you want to theme one of the out-of-the-box Chakra components?**

Let's answer both of these questions...

### Theme custom components from central theme

```jsx live=false
// 1. Use the `themeKey` option instead
const MyComponent = chakra("div", {
  themeKey: "MyComponent",
})

// 2. Move your styles to the theme under the `components.MyComponent` key
const theme = {
  components: {
    MyComponent: {
      defaultProps: {
        size: "small",
        variant: "solid",
      },
      baseStyle: { ... },
      sizes: {
        small: { ... },
        large: { ... },
      },
      variants: {
        solid: { ... },
        outline: { ... },
      },
    },
  },
}
```

### Theme out-of-the-box Chakra components from central theme

> Every out-of-the-box Chakra component has it's own themeKey available for modification in the default theme.

Reviewer: ```suggestion
Every out-of-the-box Chakra component has its own `themeKey` available for modification in the default theme.
```

There you will find `defaultProps`, `baseStyle`, `sizes`, `variants`, etc that can be tweaked as needed.

> For example, to customize the Button globally, find the `components.Button` key and modify.

Reviewer: ```suggestion
For example, to customize `Button` globally, find the `components.Button` object and modify it.
```
